File: CargoFlashCargoWebApps/JScript/Master/RegistryControl.js
var pageType = $('#hdnPageType').val();
$(document).ready(function () {
    
    //cfi.AutoComplete("AirlineSNo", "AirlineCode,AirlineName", "vwAirline", "SNo", "AirlineName", ["AirlineCode", "AirlineName"], null, "contains");
    //cfi.AutoComplete("CitySNo", "cityCode,CityName", "vcity", "SNo", "CityName", ["CityCode", "CityName"], null, "contains");
    //cfi.AutoComplete("GroupSNo", "SNo,GroupName", "vwGroups", "SNo", "GroupName", null, null, "contains");
    //cfi.AutoComplete("ProcessSNo", "SNo,ProcessName", "vwProcess", "SNo", "ProcessName", null, RegistryControlTranGrid, "contains");
    //cfi.AutoComplete("PageSNo", "SNo,PageName", "vwPage", "SNo", "PageName", null, null, "contains");

    cfi.AutoCompleteV2("AirlineSNo", "AirlineCode,AirlineName", "Master_RegistryControl_AirlineName", null, "contains");
    cfi.AutoCompleteV2("CitySNo", "cityCode,CityName", "Master_RegistryControl_CityName", null, "contains");
    cfi.AutoCompleteV2("GroupSNo", "SNo,GroupName", "Master_RegistryControl_GroupName",  null, "contains");
    cfi.AutoCompleteV2("ProcessSNo", "SNo,ProcessName", "Master_RegistryControl_ProcessName", null, RegistryControlTranGrid, "contains");
    cfi.AutoCompleteV2("PageSNo", "SNo,PageName", "Master_RegistryControl_PageName",  null, "contains");

    if (pageType != "NEW") {
        $('#Text_ProcessSNo').attr("disabled", true).closest("span").find("span").find("span.k-icon").hide();
        $('#Text_AirlineSNo').attr("disabled", true).closest("span").find("span").find("span.k-icon").hide();
        $('#Text_CitySNo').attr("disabled", true).closest("span").find("span").find("span.k-icon").hide();
        $('#Text_GroupSNo').attr("disabled", true).closest("span").find("span").find("span.k-icon").hide();
        $('#Text_PageSNo').attr("disabled", true).closest("span").find("span").find("span.k-icon").hide();
        RegistryControlTranGrid();        
    }
});

function RegistryControlTranGrid() {
    var dbTableName = 'RegistryControlTran';
    var bindvalue = $('#RecordID').val();
    if (bindvalue == '')
        bindvalue = $("#ProcessSNo").val();
    var pageType =  '0';
    cfi.ValidateForm();
    $('#tbl' + dbTableName).appendGrid({
        tableID: 'tbl' + dbTableName,
        contentEditable: pageType != 'View',
        masterTableSNo: bindvalue,
        currentPage: 1, itemsPerPage: 5, whereCondition: '', sort: '',
        servicePath: './Services/Master/RegistryControlService.svc',
        getRecordServiceMethod: 'GetRegistryControlTranRecord',
        createUpdateServiceMethod: 'createUpdate' + dbTableName,
        deleteServiceMethod: 'delete' + dbTableName,
        caption: 'Registry Control Transaction',
        initRows: 1,
        isGetRecord: true,
        columns: [
            { name: 'SPSNo', type: 'hidden' },
            { name: 'PSNo', type: 'hidden' },
            { name: 'totRowCount', type: 'hidden' },
            { name: 'SPName', display: 'SubProcess Name', type: 'label' },
            
            { name: 'priority', display: 'Priority', type: 'text', ctrlCss: { width: '50px' }, ctrlAttr: { maxlength: 3 } },
            { name: 'IsRequired', display: 'Required', type: 'checkbox' },
            { name: 'IsDisplay', display: 'Display', type: 'checkbox'},
            { name: 'IsActive', display: 'Active', type: 'checkbox' },           
            { name: 'IsOnClick', display: 'OnClick', type: 'checkbox' },
            { name: 'DC', display: 'Display Caption', type: 'text', ctrlCss: { width: '100px' }, ctrlAttr: { maxlength: 10 } },
            { name: 'ProgressCheck', display: 'Progress Check', type:'checkbox'},            
            { name: 'IsPopUpSubProcess', display: 'Popup SubProcess', type: 'checkbox' },
            { name: 'Group', display: 'Group', type: 'text', ctrlCss: { width: '100px' }, ctrlAttr: { maxlength: 100 } },
            { name: 'Status', display: 'Status', type: 'text', ctrlCss: { width: '100px' }, ctrlAttr: { maxlength: 100 } },
            

        ],
        isPaging: false,
        hideButtons: {
            remove: true,
            removeLast: true,
            insert: true,
            append: true,
            updateAll: true

        }
        
    });
    
    
}

function ExtraCondition(textId) {
    var f = cfi.getFilter("AND");
    if (textId.indexOf("PageSNo") > 0) {
        cfi.setFilter(f, "hyperlink", "neq", null)
    }
    if (textId.indexOf("ProcessSNo") > 0) {
        cfi.setFilter(f, "PageSNo", "eq", $("#PageSNo").val() == "" ? "0" : $("#PageSNo").val())
    }

    return cfi.autoCompleteFilter([f]);
}

File: CargoFlash.ROMS.Model/Master/RegistryControl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.Serialization;

namespace CargoFlash.Cargo.Model.Master
{
    [KnownType(typeof(RegistryControl))]
    public class RegistryControl
    {
        
        public Int32 SNo { get; set; }
        public string processName { get; set; }
        public string SubProcessName { get; set; }
        public string pagename { get; set; }
        public string priority { get; set; }
        public string ProcessName { get; set; }
        public string IsRequired { get; set; }

        public string Text_ProcessSNo { get; set; }
        public Int32 ProcessSNo { get; set; }

        public Int32 CitySNo { get; set; }
        public string Text_CitySNo { get; set; }

        public Int32 GroupSNo { get; set; }
        public string Text_GroupSNo { get; set; }

        public Int32 AirlineSNo { get; set; }
        public string Text_AirlineSNo { get; set; }

        public Int32 PageSNo { get; set; }
        public string Text_PageSNo { get; set; }

        public string RType { get; set; }


        public string isdisplay { get; set; }
        public string isactive { get; set; }
        public string isdeleted { get; set; }
        public string isoneclick { get; set; }
        public string displaycaption { get; set; }
        public string progresscheck { get; set; }
        public string ispopupsubprocess { get; set; }
    }

    [KnownType(typeof(RCGridTran))]
    public class RCGridTran
    {
        public string AirlineSNo {get; set;}
        public string CitySNo {get; set;}
        public string GroupSNo {get; set;}
        public string ProcessSNo { get; set; }
        public string PageSNo {get; set;}
        public string Text_AirlineSNo {get; set;}
        public string Text_CitySNo {get; set;}
        public string Text_GroupSNo {get; set;}
        public string Text_ProcessSNo {get; set;}
        public string Text_PageSNo {get; set;}
        public Int32 SPSNo { get; set; }
        public Int32 PSNo { get; set; }
        public string SPName { get; set; }
        public string priority { get; set; }
        public bool IsRequired { get; set; }
        public bool IsDisplay { get; set; }
        public bool IsActive { get; set; }
        public bool IsOnClick { get; set; }
      
        public string DC { get; set; }
        public bool ProgressCheck { get; set; }
        public bool IsPopUpSubProcess { get; set; }
        public string Status { get; set; }
        public string Group { get; set; }
        public string totRowCount { get; set; }
     
       
    }

    [KnownType(typeof(RCGridTran1))]
    public class RCGridTran1
    {
        public Int32 SPTSNo { get; set; }
        public Int32 SPSNo { get; set; }
        public string RCPriority { get; set; }
        public string priority { get; set; }
        public bool IsRequired { get; set; }
        public bool IsDisplay { get; set; }
        public bool IsActive { get; set; }
        public bool IsOnClick { get; set; }
      
        public string HdnSPSNo { get; set; }
        public bool ProgressCheck { get; set; }
        public string DisplayCaption { get; set; }


    }
}
